go/cmd/dolt/commands/checkout.go

```diff
@@ -82,237 +83,66 @@ func (cmd CheckoutCmd) Exec(ctx context.Context, commandStr string, args []strin
-		// do guess remote branch if branch.autosetupmerge is not 'false', or if it is not set, it should default to 'true'.
-		// if no remote, it should not return an error
-		remotes, err := dEnv.RepoStateReader().GetRemotes()
-		if err != nil {
-			return nil
-		}
-		remoteName, remoteBranchName = actions.ParseRemoteBranchName(startPt)
-		_, remoteOk := remotes[remoteName]
-		if !remoteOk {
-			return nil
-		}
-		verr = SetRemoteUpstreamForBranchRef(dEnv, remoteName, remoteBranchName, headRef)
-		if verr != nil {
-			return verr
-		}
-	}
-
-	return nil
-}
-
-// checkoutRemoteBranchOrSuggestNew checks out a new branch guessing the remote branch,
-// if there is a branch with matching name from exactly one remote.
-func checkoutRemoteBranchOrSuggestNew(ctx context.Context, dEnv *env.DoltEnv, name string) errhand.VerboseError {
-	remoteRefs, err := actions.GetRemoteBranchRef(ctx, dEnv.DoltDB, name)
+	sqlQuery, err := generateCheckoutSql(args)
 	if err != nil {
-		return errhand.BuildDError("fatal: unable to read from data repository.").AddCause(err).Build()
+		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usagePrt)
 	}
 
-	if len(remoteRefs) == 0 {
-		// Check if the user is trying to enter a detached head state
-		commit, _ := actions.MaybeGetCommit(ctx, dEnv, name)
-		if commit != nil {
-			// User tried to enter a detached head state, which we don't support.
-			// Inform and suggest that they check-out a new branch at this commit instead.
-
-			str := "dolt does not support a detached head state. To create a branch at this commit instead, run:\n\n" +
-				"\tdolt checkout %s -b {new_branch_name}\n"
-
-			return errhand.BuildDError(str, name).Build()
-		}
-		return errhand.BuildDError("error: could not find %s", name).Build()
-	} else if len(remoteRefs) == 1 {
-		verr := checkoutNewBranchFromStartPt(ctx, dEnv, name, remoteRefs[0].String())
-		if verr != nil {
-			return verr
-		}
-		headRef, err := dEnv.RepoStateReader().CWBHeadRef()
-		if err != nil {
-			return errhand.BuildDError(err.Error()).Build()
-		}
-		return SetRemoteUpstreamForBranchRef(dEnv, remoteRefs[0].GetRemote(), remoteRefs[0].GetBranch(), headRef)
-	} else {
-		// TODO : add hint of using `dolt checkout --track <remote>/<branch>` when --track flag is supported
-		return errhand.BuildDError("'%s' matched multiple (%v) remote tracking branches", name, len(remoteRefs)).Build()
-	}
-}
+	_, err = getRowsForSql(queryEngine, sqlCtx, sqlQuery)
 
-func checkoutNewBranchFromStartPt(ctx context.Context, dEnv *env.DoltEnv, newBranch, startPt string) errhand.VerboseError {
-	err := actions.CreateBranchWithStartPt(ctx, dEnv.DbData(), newBranch, startPt, false, nil)
-	if err != nil {
-		return errhand.BuildDError(err.Error()).Build()
+	var branchName string
+	if apr.NArg() > 0 {
+		branchName = apr.Arg(0)
 	}
-
-	return checkoutBranch(ctx, dEnv, newBranch, false)
+	return HandleVErrAndExitCode(handleErrors(branchName, err), usagePrt)
```

Comment:
The way this code reads is confusing. `HandleVErrAndExitCode(handleErrors(branchName, err)` really makes it sound like we know there was an error, and we need to handle it. I had to dig into both functions to know that this is actually the happy path. What makes this particularly awkward is that branchName was pulled out of the args to just pass it into this error handling method.

What I suggest instead is pull branchName out branch name from the args before executing the query. Then print the success message if you don't receive and error. If you do, then call it the way you are now.

Reply:
Done.